pandas/tests/indexing/test_loc.py

```diff
@@ -616,8 +612,16 @@ def test_loc_setitem_frame_with_reindex_mixed(self):
         expected["B"] = "string"
         tm.assert_frame_equal(df, expected)
 
+    def test_loc_setitem_frame_with_inverted_slice(self):
+        # GH#40480
+        df = DataFrame(index=[1, 2, 3], columns=["A", "B"], dtype=float)
+        df["B"] = "string"
+        df.loc[slice(3, 0, -1), "A"] = np.array([1, 2, 3], dtype="int64")
+        expected = DataFrame({"A": [3, 2, 1], "B": "string"}, index=[1, 2, 3])
```

Comment:
@phofl AFAICT the setting on L619 should be done in-place, so `expected["A"]` should still be float dtype.  is the changed dtype relevant here?

Reply:
ditto for test_loc_setitem_frame_with_reindex_mixed